src/raftstore/store/fsm/apply.rs

```diff
@@ -2564,30 +2575,54 @@ impl ApplyFsm {
             .force_send(Msg::LogsUpToDate(region_id));
     }
 
+    #[allow(unused_mut)]
     fn handle_snapshot(&mut self, apply_ctx: &mut ApplyContext, snap_task: GenSnapTask) {
         if self.delegate.pending_remove || self.delegate.stopped {
             return;
         }
+        let mut need_sync =
```

Comment:
How about also checking if there are pending writes?

Reply:
What are the pending writes?